### PR TITLE
feat(dlp): added a sample for dlp update stored infotype 

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp update stored infotype](https://cloud.google.com/dlp/docs/creating-stored-infotypes#rebuild) with a unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/metadata/metadata_test.go [hunk 1]
```diff
@@ -130,7 +137,7 @@ func bucketForStoredInfoType(t *testing.T, projectID string) (string, error) {
 	if !bucketExists {
 		if err := client.Bucket(bucketName).Create(ctx, projectID, &storage.BucketAttrs{
 			StorageClass: "STANDARD",
-			Location:     "us-central1",
+			Location:     "us",
```
Comment (annadata) [inline dlp/snippets/metadata/metadata_test.go:140]: Why change from us-central1 to us multi-region?

File: dlp/snippets/metadata/metadata_test.go [hunk 2]
```diff
@@ -182,3 +189,194 @@ func deleteStoredInfoTypeAfterTest(t *testing.T, name string) error {
...
+
+	fileSetUrl, gcsUri, err := bucketForUpdateStoredInfoType(t, tc.ProjectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var buf bytes.Buffer
+
+	// fileSetUrl := fmt.Sprint("gs://" + "dlp-go-lang-test" + "/" + termListFileName)
+	// gcsUri := fmt.Sprint("gs://" + "dlp-go-lang-test")
```
Comment (annadata) [inline dlp/snippets/metadata/metadata_test.go:204]: Intentional or needs cleanup?

Comment (AarshD-crest) [reply dlp/snippets/metadata/metadata_test.go:204]: Yes, Updated it.

**Review by annadata — APPROVED** [review]

File: dlp/snippets/metadata/metadata_test.go [hunk 3]
```diff
@@ -182,3 +189,192 @@ func deleteStoredInfoTypeAfterTest(t *testing.T, name string) error {
...
+	if want := "output: "; !strings.Contains(got, want) {
+		t.Errorf("error from create stored infoType %q", got)
+	}
+
+	name := strings.TrimPrefix(got, "output: ")
+
+	defer deleteStoredInfoTypeAfterTest(t, name)
+}
+
+func bucketForUpdateStoredInfoType(t *testing.T, projectID string) (string, string, error) {
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/metadata/metadata_test.go:187]: Hey, @telpirion  I have updated the code using the function from `testutil` file. Yet, I feel I am not quite sure that's what you suggested. Please check and let me know if there are any further changes.

File: dlp/snippets/metadata/metadata_test.go [hunk 4]
```diff
@@ -182,3 +189,192 @@ func deleteStoredInfoTypeAfterTest(t *testing.T, name string) error {
 	}
 	return nil
 }
+
+func TestUpdateStoredInfoType(t *testing.T) {
+	skipKOKORO(t)
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/metadata/metadata_test.go:140]: Hey Hiii @telpirion, Thank you for your feedback. After further discussions with @annadata , we have reconsidered our approach to mocking the test cases in Golang samples. Initially, we had decided to skip these test cases in the CI-CD environment. However, following yesterday's discussion with @annadata , we have now decided to remove the skip function and update these test cases accordingly.

File: dlp/snippets/metadata/metadata_test.go [hunk 5]
```diff
@@ -182,3 +189,192 @@ func deleteStoredInfoTypeAfterTest(t *testing.T, name string) error {
...
+	dirExists := false
+	query := &storage.Query{Prefix: dirPath}
+	it := client.Bucket(bucketName).Objects(ctx, query)
+	_, err = it.Next()
+	if err == nil {
+		dirExists = true
+	}
+
+	// If the directory doesn't exist, create it.
+	if !dirExists {
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/metadata/metadata_test.go]: Hey, @telpirion I have updated the code using the function from testutil file. Please check and let me know if there are any further changes.

**Review by telpirion — APPROVED** [review]

**Pull request merged**